packages/api-plugin-bull-queue/src/api/scheduleJob.js

```diff
@@ -0,0 +1,35 @@
+import { createRequire } from "module";
+import Logger from "@reactioncommerce/logger";
+
+const require = createRequire(import.meta.url);
+
+const pkg = require("../../package.json");
+
+const { name, version } = pkg;
+const logCtx = {
+  name,
+  version,
+  file: "api/scheduleJob.js"
+};
+
+/**
+ * @summary create a scheduled job
+ * @param {Object} context - The application context
+ * @param {String} queueName - The queue to add this job
+ * @param {Object} jobData - Data to be passed to the worker
+ * @param {Object} schedule - The schedule as a crontab
+ * @return {Boolean} - true if success
+ */
+export default async function scheduleJob(context, queueName, jobData, schedule) {
+  if (typeof jobData !== "object" || typeof schedule !== "string") {
+    Logger.error(logCtx, "Invalid parameters supplied to scheduleJob");
+    return false;
+  }
+  if (context.bullQueue.jobQueues[queueName]) {
+    const thisQueue = context.bullQueue.jobQueues[queueName];
+    await thisQueue.add(jobData, { repeat: { cron: schedule } });
```

Comment:
Ok, I took a slightly different tact but I think it's still less than optimal. If you add a job with an existing name, it will remove the existing job before adding a new one. I could not figure out a way to determine if the job you were adding was identical to the job that already exists so you could skip both the remove and the add. The thing that is suboptimal is that every time you restart the cluster you have a bunch of servers all removing and re-adding the job all at once with maybe some unpredictable results. Whereas before adding the same job over and over with the same schedule was a no-op.

Reply:
Yes, it does seem as though there is no great silver bullet solution. You could add a function options argument for it, such as `replaceExisting: true`, and either default that to `true` or `false` as seems best to you.